package/MDAnalysis/topology/ITPParser.py

```diff
@@ -527,7 +531,23 @@ def parse(self, include_dir='/usr/local/gromacs/share/gromacs/top/',
             self.system_molecules = [x.name for x in self._molecules]
 
         self.build_system()
-        
+
+        self.types = np.array(self.types)
+        self.charges = np.array(self.charges)
+        self.masses = np.array(self.masses)
+
+        if not all(self.charges):
+            empty = self.charges == ''
+            self.charges[empty] = [
+                (self.atomypes.get(x)["charge"] if x in self.atomypes.keys() else '') for x in self.types[empty]
```

Comment:
I am super hesitant about reading charges from atomtypes, because partial charges are so dependent on the local residue environment. Maybe we could make this a keyword argument but if there exist both an `atomtypes.atp` and an `[ atomtypes ]`, the charges of the amino acids should be taking preference. In the case where both the files exist, I think my preference would actually be

aminoacids.rtp > no charges > atomtypes

If your use case has these charges being actually accurate, I could be persuaded.

Reply:
I don't think this is an issue here, because the .atp/.rtp files are only relevant for topology building with pdb2gmx, but not when actually grompping and mdrunning. (I'm 99 % sure about that, but I stand to be corrected)

So at the point one has a topol.top file to work with, all charges have already been written to the [ atoms ] sections correctly.